project-lisa/source-code/client/src/helpers.ts

```diff
@@ -0,0 +1,45 @@
+import type { NodeishFilesystem } from "@inlang-git/fs"
+
+/**
+* Wraps a nodeishFs implementation with a js proxy for detailed logging, debugging and transparently replacing the file access behaviour.
+* advantage of using this approach is that the underlying fs can also be swapped to something like lightingfs seamlessly.
+*/
+export const wrap = (targetRoot: NodeishFilesystem, _module: string, fn?: (args: { prop: keyof typeof targetRoot, execute: () => any }) => any ): NodeishFilesystem => {
```

Comment:
can we find a better name than `wrap`?

wrap doesn't express why the fs is wrapped. `debug` maybe?